querybook/server/lib/ai_assistant/base_ai_assistant.py

```diff
@@ -50,17 +62,37 @@ def name(self) -> str:
     def set_config(self, config: dict):
         self._config = config
 
+    def catch_error(func):
```

Comment:
catch_error_to_stream?

Reply:
updated the catch_error to also handle non streaming use cases.